app/src/components/pages/ScheduleTimetable.tsx

```diff
@@ -1,26 +1,54 @@
 import React from 'react';
-import Scheduler from 'devextreme-react/scheduler';
+import { Scheduler, Editing } from 'devextreme-react/scheduler';
 import 'devextreme/dist/css/dx.light.css';
 import Appointment from 'components/organisms/Appointment';
 import classData from 'data/clean.json';
-import { parseCalendarJSON } from 'utils/utils';
+import { parseCalendarJSON_Course, parseCalendarJSON_Teacher } from 'utils/utils';
+import { Resource } from 'devextreme-react/scheduler';
 
 //The current date will be +1 month in the UI, ex: 2021/Dec/10 -> 2022/Jan/10
 const currentDate = new Date(2021, 12, 10);
 
 function ScheduleTimetable() {
-  const calendarData = parseCalendarJSON(JSON.parse(JSON.stringify(classData)));
+  const calendarData_Course = parseCalendarJSON_Course(JSON.parse(JSON.stringify(classData)));
+  const calendarData_Teacher = parseCalendarJSON_Teacher(JSON.parse(JSON.stringify(classData)));
+
   return (
-    <Scheduler
-      timeZone="Canada/Pacific"
-      dataSource={calendarData}
-      defaultCurrentView="day"
-      defaultCurrentDate={currentDate}
-      startDayHour={8}
-      height={800}
-      appointmentComponent={Appointment}
-      showAllDayPanel={false}
-    />
+    <>
+      {/*@ts-ignore*/}
+      <Scheduler
+        timeZone="Canada/Pacific"
+        dataSource={calendarData_Course}
+        views={[
+          {
+            type: 'day',
+            name: 'Day',
+            maxAppointmentsPerCell: 'auto'
+          },
+          {
+            type: 'week',
+            name: 'Week',
+            maxAppointmentsPerCell: 2
+          }
+        ]}
+        defaultCurrentView="day"
+        defaultCurrentDate={currentDate}
+        startDayHour={8}
+        height={800}
+        appointmentComponent={Appointment}
+        showAllDayPanel={false}
+      >
+        <Editing allowAdding={false} allowDragging={false} />

```

Comment:
Is this just to lock the calendar to readonly?

Reply:
Nope, we still can edit them, such as: changing time/ assign professors, etc.
For now, I just don't want dragging and adding new courses manually.